### PR TITLE
Feature/2362 state use field batch l2data of processing context

### LINKED ISSUE #2362: Improve state interface to use new field `BatchL2Data` of `ProcessingContext` struct
# Rationale

The func `ProcessAndStoreClosedBatch` have the field `encodedTxs` that is the same as `state.ProcessingContext.BatchL2Data`. This create a inconsistence in the interface. 
Related to #2357 

# Implementation
Remove `encodedTxs` of  and use the new field instead.

### PR DESCRIPTION
**Pull request by joanestebanr**
Closes #2362

### What does this PR do?

`state.ProcessAndStoreClosedBatch` use `processingCtx.BatchL2Data` instead of `encodedTxs` parameter

### Reviewers
@ARR552

### CODE REVIEW

File: synchronizer/synchronizer.go [hunk 1]
```diff
@@ -935,7 +936,8 @@ func (s *ClientSynchronizer) processSequenceBatches(sequencedBatches []etherman.
 				log.Errorf("error resetting trusted state. BatchNumber: %d, BlockNumber: %d, error: %v", batch.BatchNumber, blockNumber, err)
 				return err
 			}
-			_, flushID, proverID, err := s.state.ProcessAndStoreClosedBatch(s.ctx, processCtx, batch.BatchL2Data, dbTx, stateMetrics.SynchronizerCallerLabel)
+			processCtx.BatchL2Data = &batch.BatchL2Data
```
Comment (ARR552) [inline synchronizer/synchronizer.go]: I think this is redundant

File: state/batch.go [hunk 2]
```diff
@@ -476,7 +482,7 @@ func (s *State) ProcessAndStoreClosedBatch(ctx context.Context, processingCtx Pr
 		StateRoot:     processedBatch.NewStateRoot,
 		LocalExitRoot: processedBatch.NewLocalExitRoot,
 		AccInputHash:  processedBatch.NewAccInputHash,
-		BatchL2Data:   encodedTxs,
+		BatchL2Data:   *processingCtx.BatchL2Data,
```
Comment (ARR552) [inline state/batch.go]:
```suggestion
		BatchL2Data:   *BatchL2Data,
```
If processingCtx.BatchL2Data is nil, then will panic

**Review by ARR552 — CHANGES_REQUESTED** [review]
As a general comment, I see that this code will save twice the batchL2Data. When the batch is open and when It's closed. Do you think this could have an impact in the db performance? (I don't think so but its better to double check)

File: synchronizer/synchronizer.go (hunk 1 above)
Comment (joanestebanr) [reply synchronizer/synchronizer.go]: You are right! I'll remove this line

**Review by ARR552 — APPROVED** [review]

**Pull request closed without merging**